### PR TITLE
Bp 2848 add in3 api v3 selection to the presta shop plugin

### PR DESCRIPTION
**Pull request by vegimcarkaxhija**

### CODE REVIEW

File: buckaroo3.php [hunk 1]
```diff
@@ -968,10 +976,10 @@ public function hookPaymentOptions($params)
         if (Config::get('BUCKAROO_IN3_ENABLED') && $this->isPaymentMethodAvailable($cart, 'IN3') && $this->isIn3Available($cart)) {
             $newOption = new PaymentOption();
             $newOption->setCallToActionText($this->getBuckarooLabel('IN3', 'In3'))
-                ->setAction($this->context->link->getModuleLink('buckaroo3', 'request', ['method' => 'in3'])) // phpcs:ignore
+                ->setAction($this->context->link->getModuleLink('buckaroo3', 'request', ['method' => (new CapayableIn3())->getMethod()])) // phpcs:ignore
                 ->setInputs($this->getBuckarooFeeInputs('IN3'))
                 ->setForm($this->context->smarty->fetch('module:buckaroo3/views/templates/hook/payment_in3.tpl')) // phpcs:ignore
-                ->setLogo($this->_path . 'views/img/buckaroo_images/buckaroo_in3.png?v') // phpcs:ignore
+                ->setLogo($this->_path . 'views/img/buckaroo_images/' . (new CapayableIn3())->getLogo())
```
Comment (harli91) [inline buckaroo3.php:982]: why are you instantiating this class 3 times?

File: api/paymentmethods/in3/in3.php [hunk 2]
```diff
@@ -18,6 +18,28 @@
...
+    public $BillingPhoneNumber;
+    public $AddressesDiffer;
+    public $ShippingStreet;
+    public $ShippingHouseNumber;
+    public $ShippingHouseNumberSuffix;
+    public $ShippingPostalCode;
+    public $ShippingCity;
+    public $ShippingCountryCode;
+    public $ShippingPhoneNumber;
+    public $CustomerNumber;
```
Comment (harli91) [inline api/paymentmethods/in3/in3.php:41]: why do you use these variables and why do they start with a upper case letter?

File: classes/CapayableIn3.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $logo = Configuration::get('BUCKAROO_IN3_PAYMENT_LOGO');
+
+        if ($logo == '0') {
+            return 'buckaroo_in3.png?v';
+        }
+
+        return 'buckaroo_in3_ideal.svg?v1';
+    }
+    public function getSelectedIssuerLogo()
```
Comment (harli91) [inline classes/CapayableIn3.php:36]: in3 doesn't have issuers, where is this coming from?

File: classes/CapayableIn3.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    public function getLogo(): string
+    {
+        $logo = Configuration::get('BUCKAROO_IN3_PAYMENT_LOGO');
+
+        if ($logo == '0') {
+            return 'buckaroo_in3.png?v';
+        }
+
+        return 'buckaroo_in3_ideal.svg?v1';
```
Comment (harli91) [inline classes/CapayableIn3.php:34]: if I remember correctly if no image is selected it should default to the in3 image not the in3_ideal

**Review by harli91 — CHANGES_REQUESTED** [review]

**Review by harli91 — CHANGES_REQUESTED** [review]

File: classes/CapayableIn3.php [hunk 5]
```diff
@@ -19,25 +19,24 @@
...
 class CapayableIn3
 {
+    const VERSION_V3 = 'V3';
+    const LOGO_IN3_IDEAL = 'in3_ideal';
+    const LOGO_IN3_IDEAL_FILENAME = 'buckaroo_in3_ideal.svg?v1';
+    const LOGO_DEFAULT = 'buckaroo_in3.png?v';
     public function isV3(): bool
     {
-        return Configuration::get('BUCKAROO_IN3_API_VERSION') !== 'V2';
+        return Configuration::get('BUCKAROO_IN3_API_VERSION') === self::VERSION_V3;
```
Comment (harli91) [inline classes/CapayableIn3.php:28]: if this value is empty then we will do a v2 request as a default, the requirement was that we have v3 enabled by default

Comment (vegimcarkaxhija) [reply classes/CapayableIn3.php:28]: when plugin is installed, the value is set to V3

**Pull request merged**